doc/edgehub-api.md

```diff
@@ -0,0 +1,434 @@
+
+If Edge Hub is between IoT Hub and a device/module, Edge Hub acts as a buffer and router. It does not just relay the messages. When a device/module sends a message, Edge Hub takes it and acknowledges it even if it cannot forward the message to IoT Hub or to another module immediately. Depending on the settings, Edge Hub stores the message in a persistent store or just holds it in memory.
+
+Then Edge Hub runs the message through its routing, and depending on the routing rules it tries to forward the message to IoT Hub or another module. If it is not possible, because Edge Hub is not in connection to IoT Hub, or the module is off, then the message stays stored and will be delivered next time it is possible.
+## Twin
+[Twins](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins) are basically a combination of configuration and status information. To understand the term "Twin" and the role of this construct, let's say that we want to control a beverage cooler and we want its temperature to set 5 Celsius.
+
+In this case the part of the twin that describes the settings/configuration we want, can look the following:
+```
+"properties": {
+    "desired": {
+        "temperature": 5
+    }
+}
+```
+After the setting has been made, it might be useful to know how the device - in this case the beverage cooler - progresses achieving the desired state. To support this scenario, twins have another section where devices can report their status:
+```
+"properties": {
+    "desired": {
+        "temperature": 5
+    },
+    "reported": {
+        "temperature": 7
+    }
+}
+```
+The device can periodically report its status and many times the desired section and the reported section converges and finally take the same values. Hence the name Twin.
+
+The "desired" section is usually set by a backend application, or the operator can use Azure Portal to set the twin. The "reported" part is usually generated by the device/module.
+
+Note, that it is not necessary that the "desired" and "reported" sections have the same properties. It is possible that the "desired" section contains a property which later is not going to be reported back. Also, it can happen that a device reports status information that does not have a corresponding part in the "desired" section.
+
+From the perspective of a device/module, so far two kind of operations were mentioned:
+- reading the desired properties section
+- writing the reported properties section
+
+Iot Hub and Edge Hub allow two ways to read the desired properties:
+- Pulling the entire twin (including the desired properties section)
+- Receiving patches for desired property changes
```

Comment:
We need to add details on offline behavior - 
- if the device goes offline, if the client has subscribed to receive twin updates, EH will do a getTwin when connectivity is restored, and if there is a diff, then send a patch to the client. That way the connectivity loss is transparent to the client.